plugins/main/public/components/common/wazuh_discover/wz_discover.tsx

```diff
@@ -0,0 +1,247 @@
+  })
+
+  useEffect(() => {
+    if (!isLoading) {
+      setIndexPattern(indexPatterns?.[0] as IndexPattern);
+      search({
+        indexPattern: indexPatterns?.[0] as IndexPattern,
+        filters,
+        query,
+        pagination,
+        sorting
+      }).then((results) => {
+        setResults(results);
+        setIsSearching(false);
+      }).catch((error) => {
+        const searchError = ErrorFactory.create(HttpError, { error, message: 'Error fetching vulnerabilities' })
+        ErrorHandler.handleError(searchError);
+        setIsSearching(false);
+      })
+    }
+  }, [JSON.stringify(searchBarProps), JSON.stringify(pagination), JSON.stringify(sorting)]);
+
+  const timeField = indexPattern?.timeFieldName ? indexPattern.timeFieldName : undefined;
+
+  const onClickExportResults = async () => {
+    const params = {
+      indexPattern: indexPatterns?.[0] as IndexPattern,
+      filters,
+      query,
+      fields: columnVisibility.visibleColumns,
+      pagination: {
+        pageIndex: 0,
+        pageSize: results.hits.total
+      },
+      sorting
+    }
+    try {
+      setIsExporting(true);
+      await exportSearchToCSV(params);
+    } catch (error) {
+      const searchError = ErrorFactory.create(HttpError, { error, message: 'Error downloading csv report' })
+      ErrorHandler.handleError(searchError);
+    } finally {
+      setIsExporting(false);
+    }
+  }
+
+  return (
+    <IntlProvider locale="en">
+      <EuiPageTemplate
+        className="vulsInventoryContainer"
+        restrictWidth="100%"
+        fullHeight={true}
+        grow
+      >
+        <>
+          {isLoading ?
+            <LoadingSpinner /> :
+            <SearchBar
+              appName='inventory-vuls'
+              {...searchBarProps}
+            />}
+          {isSearching ?
+            <LoadingSpinner /> : null}
+          {!isLoading && !isSearching && results?.hits?.total === 0 ?
+            <DiscoverNoResults timeFieldName={timeField} queryLanguage={''} /> : null}
+          {!isLoading && !isSearching && results?.hits?.total > 0 ? (
+            <>
+              <EuiFlexItem grow={false}>
+                <EuiPanel hasBorder={false} hasShadow={false} color="transparent" paddingSize="none">
+                  <EuiPanel>
+                    <DashboardByRenderer
+                      input={{
+                        viewMode: ViewMode.VIEW,
+                        panels: getDiscoverPanels('wazuh-alerts-*'),
+                        isFullScreenMode: false,
+                        filters: searchBarProps.filters ?? [],
+                        useMargins: false,
+                        id: 'vulnerability-detector-dashboard-tab-filters',
+                        timeRange: {
+                          from: searchBarProps.dateRangeFrom,
+                          to: searchBarProps.dateRangeTo,
+                        },
+                        title: 'Vulnerability detector dashboard filters',
```

Comment:
The text refers to vulnerabilities but this will be used in other modules.

Reply:
Thanks @chantal-kelm